plugins/pulp_rpm/plugins/distributors/yum/metadata/updateinfo.py

```diff
@@ -16,72 +14,58 @@
 
 
 class UpdateinfoXMLFileContext(XmlFileContext):
+
     def __init__(self, working_dir, checksum_type=None, conduit=None):
         metadata_file_path = os.path.join(working_dir, REPO_DATA_DIR_NAME,
                                           UPDATE_INFO_XML_FILE_NAME)
         self.conduit = conduit
         super(UpdateinfoXMLFileContext, self).__init__(
             metadata_file_path, 'updates', checksum_type=checksum_type)
 
-    def _repo_unit_nevra(self, erratum_unit, repo_id):
+    def _get_repo_unit_nevra(self, erratum_unit, repo_nevra):
         """
         Return a list of NEVRA dicts for units in a single repo referenced by the given errata.
 
         Pulp errata units combine the known packages from all synced repos. Given an errata unit
         and a repo, return a list of NEVRA dicts that can be used to filter out packages not
-        linked to that repo when generating a repo's updateinfo XML file. While returning that
-        list of NEVRA dicts is the main goal, doing so quickly and without running out of memory
-        is what makes this a little bit tricky.
-
-        Build up a super-fancy query to get the unit ids for all NEVRA seen in these errata
-        check repo/unit associations for this errata to limit the packages in the published
-        updateinfo to the units in the repo being currently published.
+        linked to that repo when generating a repo's updateinfo XML file.
 
         :param erratum_unit: The erratum unit that should be written to updateinfo.xml.
         :type erratum_unit: pulp_rpm.plugins.db.models.Errata
-        :param repo_id: The repo_id of a pulp repository in which to find units
-        :type repo_id: str
+        :param repo_nevra: A list of NEVRA for the repo. Errata metdata only publishes for NEVRA
+                           actually contained in this repo.
+        :type repo_nevra: list of models.NEVRA objects
+
         :return: a list of NEVRA dicts for units in a single repo referenced by the given errata
         :rtype: list
         """
-        nevra_fields = ('name', 'epoch', 'version', 'release', 'arch')
-        nevra_q = mongoengine.Q()
+        repo_unit_nevra = []
```

Comment:
I think it would help readability quite a bit if we can come up with names other than ``repo_nevra`` and ``repo_unit_nevra``, since those don't help me understand how they are different.

Maybe ``nevra_in_repo`` and ``nevra_in_pkglist``, or something along those lines?